src/docfx/build/context/Context.cs

```diff
@@ -92,7 +92,7 @@ public Context(string outputPath, ErrorLog errorLog, CommandLineOptions options,
             RedirectionProvider = new RedirectionProvider(docset.DocsetPath, Config.HostName, ErrorLog, BuildScope, repositoryProvider, DocumentProvider, MonikerProvider);
             GitHubAccessor = new GitHubAccessor(Config);
             GitCommitProvider = new GitCommitProvider(repositoryProvider);
-            PublishModelBuilder = new PublishModelBuilder(outputPath, Config, Output, ErrorLog);
+            PublishModelBuilder = new PublishModelBuilder(this, outputPath, Config, Output, ErrorLog);
```

Comment:
Pass the actual dependency over the whole context.
```suggestion
            PublishModelBuilder = new PublishModelBuilder(outputPath, Config, Output, ErrorLog, MonikerProvider);
```